src/main/java/com/trillia/entities/BusinessObject.java

```diff
@@ -0,0 +1,119 @@
+package com.trillia.entities;
+
+import org.hibernate.search.annotations.*;
+
+import javax.persistence.*;
+import javax.xml.bind.annotation.XmlElement;
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Created by mmathew on 26/05/16.
+ */
+
+@Entity
+@Table(name = "trl_business_objects",uniqueConstraints = {@UniqueConstraint(columnNames = {"tenantId","objectCode"})})
+@Indexed(index = "BusinessObject")
+public class BusinessObject implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    @Column(name = "object_id", nullable = false)
+    @XmlElement(name = "object-id")
+    private long id;
+
+
+    @XmlElement(name = "tenant-id")
+    @Fields({
+            @Field,
```

Comment:
We may not need to do text search on tenantId. So the second @Field annotation should be sufficient.